x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/list_common/exception_item_card/exception_item_card.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiCommentProps } from '@elastic/eui';
+import { EuiPanel, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useMemo, useCallback } from 'react';
+import type {
+  ExceptionListItemSchema,
+  ExceptionListTypeEnum,
+} from '@kbn/securitysolution-io-ts-list-types';
+
+import { getFormattedComments } from '../../../utils/helpers';
+import * as i18n from './translations';
+import {
+  ExceptionItemCardHeader,
+  ExceptionItemCardConditions,
+  ExceptionItemCardMetaInfo,
+  ExceptionItemCardComments,
+} from '.';
+
+import type { RuleReferenceSchema, ExceptionListItemIdentifiers } from '../types';
+
+export interface ExceptionItemProps {
+  dataTestSubj?: string;
+  disableActions?: boolean;
+  exceptionItem: ExceptionListItemSchema;
+  listType: ExceptionListTypeEnum;
+  ruleReferences: RuleReferenceSchema[];
+  editActionLabel?: string;
+  deleteActionLabel?: string;
+  onDeleteException: (arg: ExceptionListItemIdentifiers) => void;
+  onEditException: (item: ExceptionListItemSchema) => void;
+}
+
+const ExceptionItemCardComponent = ({
+  disableActions = false,
+  exceptionItem,
+  listType,
+  ruleReferences,
+  dataTestSubj,
+  editActionLabel,
+  deleteActionLabel,
+  onDeleteException,
+  onEditException,
+}: ExceptionItemProps): JSX.Element => {
+  const handleDelete = useCallback((): void => {
+    onDeleteException({
+      id: exceptionItem.id,
+      name: exceptionItem.name,
+      namespaceType: exceptionItem.namespace_type,
+    });
+  }, [onDeleteException, exceptionItem.id, exceptionItem.name, exceptionItem.namespace_type]);
+
+  const handleEdit = useCallback((): void => {
+    onEditException(exceptionItem);
+  }, [onEditException, exceptionItem]);
+
+  const formattedComments = useMemo((): EuiCommentProps[] => {
+    return getFormattedComments(exceptionItem.comments);
+  }, [exceptionItem.comments]);
+
+  const actions = useMemo(
+    () => [
+      {
+        key: 'edit',
+        icon: 'controlsHorizontal',
+        label: (editActionLabel || i18n.exceptionItemCardEditButton(listType)) as string,
```

Comment:
Weird that it gets typed as `any` without this. When I add a return type to `exceptionItemCardEditButton`, it fixes it so the cast isn't needed anymore.
```
export const exceptionItemCardEditButton = (listType: string): string =>
```

Reply:
I got resolved even without the casting `as string` when I fixed the translation file